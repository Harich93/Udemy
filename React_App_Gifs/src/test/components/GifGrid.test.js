import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")


describe('Pruebas en el <GifGrid />', () => {
    
    const categoria = 'Goku';

    test('Snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
         }); 

        const wrapper = shallow( <GifGrid categoria={ categoria }  /> );
         
        expect( wrapper ).toMatchSnapshot(); 
    });

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
       
        const gifs = [{
            id: 'abc',
            url: 'http://localhost/cualquiercosa.jpg',
            title: 'Lo que sea'
        }]

        useFetchGifs.mockReturnValue({
           data: gifs,
           loading: false
        }); 

        const wrapper = shallow( <GifGrid categoria={ categoria }  /> );

        expect( wrapper.find('p').exists() ).toBe( false ); 
        expect( wrapper.find('GrfGridItem').length ).toBe( gifs.length );
        
    });

})
