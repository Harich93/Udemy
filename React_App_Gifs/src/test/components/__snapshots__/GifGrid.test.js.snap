// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInTopLeft"
  >
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
